### PR TITLE
feat(discord): Import single channel from discord

### PR DESCRIPTION
**Pull request by borismelnik**
Fixes: #12238

### What does the PR do

Importing single channel from discord functionality 

### Affected areas

Discord import 

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [ ] I've checked the design and this PR matches it

Снимок экрана 2023-10-16 в 21 40 26
Снимок экрана 2023-10-16 в 21 41 20

### CODE REVIEW

File: ui/app/AppLayouts/Communities/stores/CommunitiesStore.qml [hunk 1]
```diff
@@ -239,7 +240,9 @@ QtObject {
                                              // TODO
                                          }
                                       }, from = 0) {
-        console.warn("!!! IMPLEMENT ME requestImportDiscordChannel") // FIXME
+        communitiesModuleInst.requestImportDiscordChannel(args.name, args.discordChannelId, args.communityId,
+                                                        args.description, args.color, args.emoji, from)
+//        console.warn("!!! IMPLEMENT ME requestImportDiscordChannel") // FIXME
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/stores/CommunitiesStore.qml:245]: Can be removed :)

File: src/app/modules/main/communities/io_interface.nim [hunk 2]
```diff
@@ -135,12 +140,15 @@ method requestExtractDiscordChannelsAndCategories*(self: AccessInterface, filesT
 method discordCategoriesAndChannelsExtracted*(self: AccessInterface, categories: seq[DiscordCategoryDto], channels: seq[DiscordChannelDto], oldestMessageTimestamp: int, errors: Table[string, DiscordImportError], errorsCount: int) {.base.} =
   raise newException(ValueError, "No implementation available")
 
-method discordImportProgressUpdated*(self: AccessInterface, communityId: string, communityName: string, communityImage: string, tasks: seq[DiscordImportTaskProgress], progress: float, errorsCount: int, warningsCount: int, stopped: bool, totalChunksCount: int, currentChunk: int) {.base.} =
+method discordImportProgressUpdated*(self: AccessInterface, communityId: string, communityName: string, channelId: string, channelName: string, communityImage: string, tasks: seq[DiscordImportTaskProgress], progress: float, errorsCount: int, warningsCount: int, stopped: bool, totalChunksCount: int, currentChunk: int) {.base.} =
```
Comment (caybro) [inline src/app/modules/main/communities/io_interface.nim:143]: I don't see this being updated in `module.nim`; it still has this signature:
```nim
method discordImportProgressUpdated*(
    self: Module,
    communityId: string,
    communityName: string,
    communityImage: string,
    tasks: seq[DiscordImportTaskProgress],
    progress: float,
    errorsCount: int,
    warningsCount: int,
    stopped: bool,
    totalChunksCount: int,
    currentChunk: int
```

File: src/app_service/service/community/service.nim [hunk 3]
```diff
@@ -1024,6 +1028,34 @@ QtObject:
...
+          emoji,
+          filesToImport,
+          fromTimestamp)
+
+        if response.error != nil:
+          let error = Json.decode($response.error, RpcError)
+          raise newException(RpcException, "Error importing discord channel: " & error.message)
+
+      except Exception as e:
+        error "Error importing discord channel", msg = e.msg
```
Comment (jrainville) [inline src/app_service/service/community/service.nim:1057]: Do we have an error state in the popup for a failure to start?

**Review by jrainville — APPROVED** [review]
Looks good

File: src/app/modules/main/communities/io_interface.nim (hunk 2 above)
Comment (caybro) [reply src/app/modules/main/communities/io_interface.nim:143]: Ah my bad!

**Review by caybro — APPROVED** [review]
LGTM

**Review by glitchminer — CHANGES_REQUESTED** [review]
Hi @borismelnik , I have not successfully been able to import messages. I'm using test data that was shared but modified to have only 1 message, the earliest one.

First test fails to complete due to failed asset download. 
image

---
Second test with same data but the failing asset removed fails to complete or create channel at all. App becomes unresponsive and shows messages imported as 133%
image

---
Third test using full unmodified json also failed in the same way as the second.

**Comment by borismelnik** [comment]
Moved to https://github.com/status-im/status-desktop/pull/12634

**Pull request closed without merging**